pipeline_dp/pipeline_operations.py

```diff
@@ -349,3 +346,213 @@ def count_per_element(self, col, stage_name: typing.Optional[str] = None):
+            keys = set(k for k, v in self.job_inputs)
+            self.results_dict.update({k: self.manager.list() for k in keys})
+            self._init_pool().map(_pool_worker, self.job_inputs, self.chunksize)
+            self._outputs = ((k, list(v)) for k, v in self.results_dict.items())
+
+class _LazyMultiProcCountIterator(_LazyMultiProcIterator):
+    def __init__(self, job_inputs: typing.Iterable,
+                chunksize: int,
+                n_jobs: typing.Optional[int],
+                **pool_kwargs):
+        self.manager = mp.Manager()
+        self.results_dict = self.manager.dict()
+        def insert_row(captures, key):
+            (results_dict_,) = captures
+            results_dict_[key] += 1
+
+        insert_row = partial(insert_row, (self.results_dict,))
+
+        super().__init__(insert_row, job_inputs, 
+                         chunksize=chunksize, 
+                         n_jobs=n_jobs, 
+                         **pool_kwargs)
+
+    def _trigger_iterations(self):
+        if self._outputs is None:
+            keys = set(self.job_inputs)
+            self.results_dict.update({k: 0 for k in keys})
+            self._init_pool().map(_pool_worker, self.job_inputs, self.chunksize)
+            self._outputs = self.results_dict.items()
+
+
+class MultiProcLocalPipelineOperations(PipelineOperations):
+    def __init__(
+        self, n_jobs: typing.Optional[int] = None, chunksize: int = 1, **pool_kwargs
+    ):
+        self.n_jobs = n_jobs
+        self.chunksize = chunksize
+        self.pool_kwargs = pool_kwargs
+
+    def map(self, col, fn, stage_name: typing.Optional[str] = None):
+        return _LazyMultiProcIterator(job=fn, job_inputs=col, 
+                                      chunksize=self.chunksize, 
+                                      **self.pool_kwargs)
+
+    def flat_map(self, col, fn, stage_name: typing.Optional[str] = None):
+        return (e for x in self.map(col, fn, stage_name) for e in x)
+
+    def map_tuple(self, col, fn, stage_name: typing.Optional[str] = None):
+        return self.map(col, lambda row: fn(*row), stage_name)
+
+    def map_values(self, col, fn, stage_name: typing.Optional[str] = None):
+        return self.map(col, lambda x: (x[0], fn(x[1])), stage_name)
+
+    def group_by_key(self, col, stage_name: typing.Optional[str] = None):
+        # NOTE - this cannot be implemented in an ordered manner without (almost) serial execution!
+        #   both keys and groups will be out of order
+        return _LazyMultiProcGroupByIterator(col, self.chunksize, 
+                                             self.n_jobs, 
+                                             **self.pool_kwargs)
+
+    def filter(self, col, fn, stage_name: typing.Optional[str] = None):
+        ordered_predicates = self.map(col, fn, stage_name)
+        return (row for row, keep in zip(col, ordered_predicates) if keep)
+
+    def filter_by_key(self, col,
+                      public_partitions,
```

Comment:
nit: It's a little weird to be referring to public partitions in a "filter by key" operation, since this is at a lower level than the concept of "partitions."

Reply:
This is part of the API that was specified by the `PipelineOperations` abstract base class.